### PR TITLE
openmw: 0.46 -> 0.47

### PR DESCRIPTION
**Pull request by marius851000**
###### Motivation for this change

OpenMW version 0.47 was released, with some really interesting new feature (was using a nightly version compiled via Nix previously, on which this commit is based)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

note : 
openmw dropped support for Qt4 (so removed the compilation flag)
it add two dependancies, a direct one on recastnavigation, and an indirect on OSG with collada (adding some support for collada format)
The optional opencollada depencie of OSG was broken. I repeared it, using the code referenced in the OpenMW CI.
Added myself to OpenMW maintainers.

### CODE REVIEW

**Comment by marius851000** [comment]
openmw maintainer : @abbradar 
openscenegraph maintainer (fixing collada support) : @aanderse @7c6f434c

**Review by 7c6f434c — APPROVED** [review]
Looks fine (not that I know what Collada does, but the changes seem to make sense)

**Review by aanderse — APPROVED** [review]
LGTM :+1: Thanks!

**Comment by marius851000** [comment]
Updated the OpenMW bullet to fix physics mulithreading. I will see if I should update the global bullet package, but that will for another PR.

File: pkgs/development/libraries/collada-dom/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, fetchFromGitHub, cmake, boost, libxml2, minizip, readline }:
+
+stdenv.mkDerivation {
+  pname = "collada-dom";
+  version = "rdiankov-2020-01-03";
```
Comment (thiagokokada) [inline pkgs/development/libraries/collada-dom/default.nix:5]:
```suggestion
  version = "unstable-2020-01-03";
```

File: pkgs/development/libraries/recastnavigation/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+stdenv.mkDerivation rec {
+  pname = "recastai";
+  # use latest revision for the CMake build process and OpenMW
+  # OpenMW use e75adf86f91eb3082220085e42dda62679f9a3ea
+  version = "2021-03-05";
+
+  src = fetchFromGitHub {
+    owner = "recastnavigation";
+    repo = "recastnavigation";
+    rev = "c5cbd53024c8a9d8d097a4371215e3342d2fdc87";
```
Comment (thiagokokada) [inline pkgs/development/libraries/recastnavigation/default.nix:12]: Checks: https://github.com/recastnavigation/recastnavigation/commit/c5cbd53024c8a9d8d097a4371215e3342d2fdc87

File: pkgs/development/libraries/recastnavigation/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, cmake, libGL, SDL2, libGLU }:
+
+stdenv.mkDerivation rec {
+  pname = "recastai";
+  # use latest revision for the CMake build process and OpenMW
+  # OpenMW use e75adf86f91eb3082220085e42dda62679f9a3ea
+  version = "2021-03-05";
```
Comment (thiagokokada) [inline pkgs/development/libraries/recastnavigation/default.nix:7]:
```suggestion
  version = "unstable-2021-03-05";
```

File: pkgs/games/openmw/default.nix [hunk 4]
```diff
@@ -13,56 +13,75 @@
 , unshield
 , openal
 , libXt
+, lz4
+, recastnavigation
 }:
 
 let
-  openscenegraph_ = openscenegraph.overrideDerivation (self: {
+  openscenegraph_openmw = (openscenegraph.override {colladaSupport = true; })
```
Comment (thiagokokada) [inline pkgs/games/openmw/default.nix:21]:
```suggestion
  openscenegraph_openmw = (openscenegraph.override { colladaSupport = true; })
```

**Review by thiagokokada — COMMENTED** [review]
Almost good, just some fixes.

File: pkgs/development/libraries/collada-dom/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner = "rdiankov";
+    repo = "collada-dom";
+    rev = "c1e20b7d6ff806237030fe82f126cb86d661f063";
```
[the remark this answers is not in the document]
Comment (marius851000) [reply pkgs/development/libraries/collada-dom/default.nix:10]: check what ? I don't understand.

[the remark this answers is not in the document]
Comment (thiagokokada) [reply pkgs/development/libraries/collada-dom/default.nix:10]: It is just that I checked this commit matches a commit from the repository.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 144798` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>openmw-tes3mp</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>collada-dom</li>
    <li>openmw</li>
    <li>recastnavigation</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Is `openmw-tes3mp` failure something that is supposed to happen?

**Comment by 1MachineElf** [comment]
This OpenMW 0.47 nix package seems to work for me. :)

**Comment by marius851000** [comment]
Oops (for tes3mp not compiling). Here is what happen I forget to run nixpkgs-review... Will take a look at it back at home.

File: pkgs/games/openmw/default.nix [hunk 6]
```diff
@@ -13,56 +13,75 @@
...
   cmakeFlags = [
-    "-DDESIRED_QT_VERSION:INT=5"
     # as of 0.46, openmw is broken with GLVND
     "-DOpenGL_GL_PREFERENCE=LEGACY"
+    "-DOPENMW_USE_SYSTEM_RECASTNAVIGATION=1"
   ];
 
   meta = with lib; {
     description = "An unofficial open source engine reimplementation of the game Morrowind";
     homepage = "http://openmw.org";
```
Comment (SuperSandro2000) [inline pkgs/games/openmw/default.nix:82]:
```suggestion
    homepage = "https://openmw.org/";
```

File: pkgs/games/openmw/default.nix [hunk 7]
```diff
@@ -13,56 +13,75 @@
...
+      "-DUSE_DOUBLE_PRECISION=ON"
+      "-DBULLET2_MULTITHREADING=ON"
+    ];
   });
 
 in
 mkDerivation rec {
-  version = "0.46.0";
+  version = "0.47.0";
   pname = "openmw";
```
Comment (SuperSandro2000) [inline pkgs/games/openmw/default.nix:49]:
```suggestion
  pname = "openmw";
  version = "0.47.0";
```

File: pkgs/games/openmw/default.nix [hunk 8]
```diff
@@ -13,56 +13,75 @@
...
+      src = fetchFromGitHub {
+        owner = "OpenMW";
+        repo = "osg";
+        rev = "bbe61c3bc510a4f5bb4aea21cce506519c2d24e6";
+        sha256 = "sha256-t3smLqstp7wWfi9HXJoBCek+3acqt/ySBYF8RJOG6Mo=";
+      };
+    });
+
+  bullet_openmw = bullet.overrideDerivation (old: rec {
+    version = "3.17";
```
Comment (SuperSandro2000) [inline pkgs/games/openmw/default.nix:32]: Can we update bullet or add buller_3 instead?

File: pkgs/development/libraries/collada-dom/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "A1ne/D6S0shwCzb9spd1MoSt/238HWA8dvgd+DC9cXc=";
+  };
+
+  postInstall = ''
+    chmod +w -R $out
+    mv $out/include/*/* $out/include
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/collada-dom/default.nix:16]: Maybe we should symlink to not break imports of other packages?

File: pkgs/development/libraries/collada-dom/default.nix [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+stdenv.mkDerivation {
+  pname = "collada-dom";
+  version = "unstable-2020-01-03";
+
+  src = fetchFromGitHub {
+    owner = "rdiankov";
+    repo = "collada-dom";
+    rev = "c1e20b7d6ff806237030fe82f126cb86d661f063";
+    sha256 = "A1ne/D6S0shwCzb9spd1MoSt/238HWA8dvgd+DC9cXc=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/collada-dom/default.nix:11]:
```suggestion
    sha256 = "sha256-A1ne/D6S0shwCzb9spd1MoSt/238HWA8dvgd+DC9cXc=";
```

**Comment by SuperSandro2000** [comment]
thanks @thiagokokada and the others. PR is already in a really good state!

File: pkgs/games/openmw/default.nix (hunk 8 above)
Comment (marius851000) [reply pkgs/games/openmw/default.nix:32]: I don't know. I'll probably take a look at that later, but seeing there are other software that depend on bullet, and that it seems to be a major new version, I plan to make another PR dedicated to this.

Comment (marius851000) [reply pkgs/games/openmw/default.nix:32]: Well, it seems updating the library doesn't cause any build failure nor cause runtime failure on vdrift or stuntrally. I'll open a different PR for it,

Comment (marius851000) [reply pkgs/games/openmw/default.nix:32]: Done here : https://github.com/NixOS/nixpkgs/pull/145385

**Comment by marius851000** [comment]
So, it seems that my checks (https://github.com/NixOS/nixpkgs/pull/145385) confirm that tes3mp compiled on master...

**Comment by thiagokokada** [comment]
> So, it seems that my checks (#145385) confirm that tes3mp compiled on master...

Just to understand here, we should merge #145385 and afterwards this one?

**Comment by marius851000** [comment]
Oh. I just meant that it fetched tes3mp from the hydra cache and that it compiled, so I should broken it somehow here (it's compiling right now on this branch)

**Comment by marius851000** [comment]
So, I fixed tes3mp compilation. But I have issue going on a public server (due to a version mismatch). I'll have to fix this later (but doesn't seems worse than the current version). Also added a small patch that would allow ``tes3mp-browser`` to correctly locate the binary (instead of using the one in the current folder).

**Comment by thiagokokada** [comment]
> So, I fixed tes3mp compilation. But I have issue going on a public server (due to a version mismatch). I'll have to fix this later (but doesn't seems worse than the current version). Also added a small patch that would allow `tes3mp-browser` to correctly locate the binary (instead of using the one in the current folder).

Do you want to fix this in this PR? Or can I go and merge it as-is (since it isn't worse than it current status)?

**Comment by marius851000** [comment]
I'll probably take a look later, and I'll prefer to make another PR for fixing the tes3mp version.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 144798` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>collada-dom</li>
    <li>openmw</li>
    <li>openmw-tes3mp</li>
    <li>portmod</li>
    <li>recastnavigation</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Thanks folks. Merged.

Future improvements can come in other PRs.

**Pull request merged**